File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Set<Integer> set = new HashSet<>();
        int cur = 0,pre = 0;
        for(int a:nums){
            cur = (cur + a) % k;
            if(set.contains(cur)) return true;
            set.add(pre);
            pre = cur;
        }
        return false;
    }
}

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        prefix={0:-1}
        curr=0
        for i in range(len(nums)):
            curr+=nums[i]
            if curr%k in prefix:
                if prefix[curr%k]<=i-2:
                    return True
            else:
                prefix[curr%k]=i
        return False